### ISSUE TITLE
Restart Function

### ISSUE THREAD

**Issue by markcmarino** [inkle/ink#236]
Is there hope for a method to restart a game and erase the history?  That would be of use to many Ink projects.

**Comment by facelesspanda** [comment]
If there is an "init" phase somewhere in your game that initializes everything in its ink objects, wouldn't calling that again reinitialize everything?

**Comment by joethephish** [comment]
I believe `Story.ResetState()` should be exactly what you're looking for?

**Comment by markcmarino** [comment]
Is there anything within the Ink language that can be called?

**Comment by joethephish** [comment]
Ah, nope. ink is designed primarily as a scripting language to use within a wider game, so you'd be expected to make a game-side call. What's your use case?

**Comment by joningold** [comment]
No, you'd need to hook it up to an external function to restart it. That
should be possible in the JS export version as well; just remember to do
"story exports" only so you don't overwrite your js code!!

cheers
jon

**Comment by joethephish** [comment]
I wouldn't recommend resetting the story directly from within an external function, since when an external function is being called, it's in the middle of evaluating the ink flow. So it would probably do very odd/funky things indeed! You could safely set some variables to zero though.
